pymc_marketing/mlflow.py

```diff
@@ -412,14 +463,344 @@ def log_inference_data(
+        data = pd.read_csv(data_url, parse_dates=["date_week"])
+
+        X = data.drop("y",axis=1)
+        y = data["y"]
+
+        mmm = MMM(
+            adstock=GeometricAdstock(l_max=8),
+            saturation=LogisticSaturation(),
+            date_column="date_week",
+            channel_columns=["x1", "x2"],
+            control_columns=[
+                "event_1",
+                "event_2",
+                "t",
+            ],
+            yearly_seasonality=2,
+        )
+
+        mlflow.set_experiment("MMM Experiment")
+
+        with mlflow.start_run():
+            idata = mmm.fit(X, y)
+
+            # Additional specific logging
+            fig = mmm.plot_components_contributions()
+            mlflow.log_figure(fig, "components.png")
+
+            model_info = log_model(mmm,
+                    registered_model_name="my_amazing_mmm",
+                    include_last_observations=True,
+                    original_scale=False,
+                    )
+    """
+    # Incorporate MMM into MLflow workflow
+    mlflow_mmm = MMMWrapper(
+        mmm,
+        extend_idata=wrapper_kwargs.get("extend_idata", False),
+        combined=wrapper_kwargs.get("combined", True),
+        include_last_observations=wrapper_kwargs.get(
+            "include_last_observations", False
+        ),
+        original_scale=wrapper_kwargs.get("original_scale", True),
+    )
+
+    mlflow.pyfunc.log_model(
+        artifact_path=artifact_path,
+        python_model=mlflow_mmm,
+    )
+    run_id = mlflow.active_run().info.run_id
+    model_uri = f"runs:/{run_id}/{artifact_path}"
+
+    if registered_model_name:
+        mlflow.register_model(model_uri, registered_model_name)
+
+
+def load_model(
+    run_id: str,
+    full_model: bool = False,
+    keep_idata: bool = False,
+    artifact_path: str = "model",
+    dst_path: str | None = None,
+) -> mlflow.pyfunc.PyFuncModel | MMM:
+    """
+    Load a PyMC-Marketing MMM model from MLflow.
+
+    Can either load the full model including the InferenceData, or just the lighter PyFuncModel version.
+
+    Parameters
+    ----------
+    run_id : str
+        The MLflow run ID from which to load the model.
+    full_model : bool, default=True
+        If True, load the full MMM model including the InferenceData.
+    keep_idata : bool, default=False
+        If True, keep the downloaded InferenceData saved locally.
+    artifact_path : str, default="model"
+        The artifact path within the run where the model is stored.
+    dst_path : str | None, default=None
+        The local destination path where the InferenceData will be downloaded.
+        If None, defaults to "idata_{run_id}" to avoid conflicts when loading multiple models.
+
+    Returns
+    -------
+    model : mlflow.pyfunc.PyFuncModel | MMM
+        The loaded MLflow PyFuncModel or MMM model.
+
+
+    Examples
+    --------
+    # Load model using run_id
+    model = load_model(run_id="your_run_id", full_model=True, keep_idata=True)
```

Comment:
Wrap this in the code-block like in the other examples. Required to render in the documentation